test/core/ChainManager.test.js

```diff
@@ -0,0 +1,131 @@
+import SocketService from '../../src/core/SocketService';
+import ChainManager from '../../src/core/ChainManager';
+import {
+  listAllChains,
+  listAllSnapshots,
+  getBlockNumber,
+  mineBlock
+} from '../../src/core/ChainRequest';
+
+jest.setTimeout(10000);
+let socket, service, id, chain, exists, response, block;
```

Comment:
something should be declared globally only if it has to have its value preserved between blocks (i.e. from beforeEach to a test, or between tests). if not, declare it within the test, even if it seems repetitive. this seems to apply to `exists` at least

Reply:
in fact i don't think `exists` is needed at all; you can be doing
```js
expect(await service.exists(id)).toBeTruthy();
```

Reply:
Had thought for some reason that you couldn't pass `await` in a expect for some reason.